### PR TITLE
Team1 API Spec

### PR DESCRIPTION
**Pull request by SimranMakhija7**

### CODE REVIEW

File: api-spec.md [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+| :----------------------------------------------------- | :----: | :------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------ |
+| /config                                                |  GET   | All REST clients should first call this route to get catalog configuration properties from the server to configure the catalog and its HTTP client.Configuration from the server consists of two sets of key/value pairs.                                                                                                                                                                                                                                                                                                                   |
```
Comment (zhouzilong2020) [inline api-spec.md:7]: Should we hide this from outside? only the catalog client code can access this.

File: api-spec.md [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+| /namespaces/{namespace}/tables/{table}                 |  GET   | summary: Load a table from the catalog operationId: loadTable description: Load a table from the catalog.The response contains both configuration and table metadata. The configuration, if non-empty is used as additional configuration for the table that overrides catalog configuration. For example, this configuration may change the FileIO implementation to be used for the table. The response also contains the table's full metadata, matching the table metadata JSON file.                                                   |
+| /namespaces/{namespace}/tables/{table}                 |  POST  | ??                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                          |
```
Comment (zhouzilong2020) [inline api-spec.md:18]: updating table?
e.g. dumping more data to table

**Review by zhouzilong2020 — COMMENTED** [review]
While the design seems reasonable, I'm uncertain whether we should make the HTTP API accessible to other teams or limit access solely to our Rust client code.
our team basically did the same thing as yours.

File: api-spec.md (hunk 1 above)
Comment (Aditya-06) [reply api-spec.md:7]: Here, we are just exposing the configurations of the catalog that anyone can read. Something that Iceberg did just for the purpose of displaying information. Whether we choose to hide/expose could be a design decision.

File: api-spec.md (hunk 2 above)
Comment (Aditya-06) [reply api-spec.md:18]: This was a point of contention as mentioned in the Piazza post. Not sure if this is something we can ignore for now

**Comment by SimranMakhija7** [comment]
> limit access solely to our Rust client code

what do you mean by rust client code?

**Comment by zhouzilong2020** [comment]
> > limit access solely to our Rust client code
> 
> what do you mean by rust client code?

If we only expose REST API to other teams, they may have to write their own client code to interact with us.
We could codegen the RUST API client and expose that to other teams.

**Comment by SimranMakhija7** [comment]
oh ok. Do the API endpoints look good? Do you have anything else you'd like to add?

**Comment by zhouzilong2020** [comment]
> oh ok. Do the API endpoints look good? Do you have anything else you'd like to add?

yep this looks good to me.
our team's just use the openapi and strip off authentication, view and transaction.
https://github.com/cmu-db/15721-s24-catalog2/blob/main/doc/openapi/rest-catalog-open-api.yaml

**Pull request merged**